src/classes/BusinessLogicServices/CarrierService.php

```diff
@@ -495,6 +495,33 @@ private function setCarrierRange(\Carrier $carrier, ShippingMethod $shippingMeth
         return $range;
     }
 
+    /**
+     * Removes carrier ranges, if any.
+     *
+     * @param \Carrier $carrier
+     *
+     * @throws \PrestaShopDatabaseException
+     * @throws \PrestaShopException
+     */
+    private function removeCarrierRanges(\Carrier $carrier)
+    {
+        $priceRanges = \RangePrice::getRanges($carrier->id);
+        if ($priceRanges) {
+            foreach ($priceRanges as $priceRangeData) {
+                $range = new \RangePrice($priceRangeData['id_range_price']);
+                $range->delete();
+            }
+        }
+
+        $weightRanges = \RangeWeight::getRanges($carrier->id);
+        if ($weightRanges) {
```

Comment:
Why do you need to repeat the same control flow structure?

Reply:
Because different classes are used and different ID keys (`RangePrice` and `RangeWeight`). Unfortunately, PrestaShop has this separated. Extracting a method for both would require some magic methods like calling a static method for arbitrary class.

Reply:
I don't know if is posibble to implement High-order functions in PHP but I think with this approach you could solve this problem

Reply:
Added some magic.